src/index.ts

```diff
@@ -43,13 +41,45 @@ import * as path from "path";
 import { MetalsFeatures } from "./MetalsFeatures";
 import DecorationProvider from "./decoration";
 import { InputBoxOptions } from "./portedProtocol";
+import { TreeViewController } from "./tvp/controller";
+import { TreeViewFeature } from "./tvp/feature";
+import { TreeViewsManager } from "./tvp/treeviews";
 
 export async function activate(context: ExtensionContext) {
   detechLauncConfigurationChanges();
   await checkServerVersion();
 
+  const mbServiceStat = services
+    .getServiceStats()
+    .find(stat => stat.languageIds.indexOf("scala") != -1);
+  if (mbServiceStat === undefined) return;
+  const metalsClient = services.getService(mbServiceStat.id).client;
```

Comment:
Maybe I use different configuration than you use. I have following lines in my file coc-settings.json:
```
{
  "languageserver": {
    "metals": {
      "command": "metals-vim",
      "rootPatterns": ["build.sbt"],
      "filetypes": ["scala", "sbt"]
    }
  }
}
```
So coc.nvim starts metals-vim client automatically. By some reason you start another second client which is not registered in coc.nvim at all. I don't understand why this client is needed.

Reply:
Registering the client in that manner doesn't make a ton of sense, since it sort of defeats the purpose of having the extension register the client. If you take a look at the coc-tsserver, coc-python, or really any of the other extensions, they all do it this way. It's done purposefully so the user doesn't have manually do it. For example, the goal is to simply have a user issue a `:CocInstall coc-metals`, and that does everything they need to get started. Again, this is how it works in the other coc extensions as well and how I think we should approach it. 

Have you by chance used `coc-metals` a bit before implementing this? If not, it may be a good idea to use it a bit to see how things work without having to manually register metals, and I think you'll get the idea of what that's done.

Reply:
OK, I removed most of destructive changes in index.ts file :) Hopefully it should work now in your environment.

Reply:
Haha thank you. I was able to pull this down and use it a bit. This is really incredible. Thanks a ton for your work. I'll keep playing around with things and give you some initial impressions and some more thoughts on the code.

This is exciting!